src/Release/Version.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+declare(strict_types=1);
+
+namespace phpweb\Release;
+
+final readonly class Version
+{
+    private function __construct(
+        private Major $major,
+        private Minor $minor,
+        private Patch $patch,
+    ) {
+    }
+
+    public static function create(
+        Major $major,
+        Minor $minor,
+        Patch $patch,
+    ): self {
+        return new self(
+            $major,
+            $minor,
+            $patch,
+        );
+    }
+
+    /**
+     * @throws \InvalidArgumentException
+     */
+    public static function fromString(string $value): self
+    {
+        if (1 !== preg_match('/^(?P<major>(0|[1-9]\d*))\.(?P<minor>(0|[1-9]\d*))\.(?P<patch>(0|[1-9]\d*))$/', $value, $matches)) {
+            throw new \InvalidArgumentException(\sprintf(
+                'Value "%s" does not appear to be a valid value for a version.',
+                $value
+            ));
```

Comment:
```suggestion
            throw new \ValueError(\sprintf(
                'Value "%s" does not appear to be a valid value for a version.',
                $value
            ));
```
We could just throw a ValueError